test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionProtocolTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.AspNetCore.SignalR.Tests.Common;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Client.Tests
+{
+    // This includes tests that verify HubConnection conforms to the Hub Protocol, without setting up a full server (even TestServer).
+    // We can also have more control over the messages we send to HubConnection in order to ensure that protocol errors and other quirks
+    // don't cause problems.
+    public class HubConnectionProtocolTests
+    {
+        [Fact]
+        public async Task InvokeSendsAnInvocationMessage()
+        {
+            var connection = new TestConnection();
+            var hubConnection = new HubConnection(connection, new JsonHubProtocol(new JsonSerializer()), new LoggerFactory());
+            try
+            {
+                var invokeTask = hubConnection.Invoke("Foo", typeof(void));
```

Comment:
Good point, I had some code in TestConnection that required calling StartAsync at one point, I'll put that in because it's best to emulate the same call sequence.